lib/rack.rb

```diff
@@ -25,67 +25,6 @@ def self.release
-  HTTPS             = 'HTTPS'.freeze
-  PATH_INFO         = 'PATH_INFO'.freeze
-  REQUEST_METHOD    = 'REQUEST_METHOD'.freeze
-  REQUEST_PATH      = 'REQUEST_PATH'.freeze
-  SCRIPT_NAME       = 'SCRIPT_NAME'.freeze
-  QUERY_STRING      = 'QUERY_STRING'.freeze
-  SERVER_PROTOCOL   = 'SERVER_PROTOCOL'.freeze
-  SERVER_NAME       = 'SERVER_NAME'.freeze
-  SERVER_ADDR       = 'SERVER_ADDR'.freeze
-  SERVER_PORT       = 'SERVER_PORT'.freeze
-  CACHE_CONTROL     = 'Cache-Control'.freeze
-  CONTENT_LENGTH    = 'Content-Length'.freeze
-  CONTENT_TYPE      = 'Content-Type'.freeze
-  SET_COOKIE        = 'Set-Cookie'.freeze
-  TRANSFER_ENCODING = 'Transfer-Encoding'.freeze
-  HTTP_COOKIE       = 'HTTP_COOKIE'.freeze
-  ETAG              = 'ETag'.freeze
-
-  # HTTP method verbs
-  GET     = 'GET'.freeze
-  POST    = 'POST'.freeze
-  PUT     = 'PUT'.freeze
-  PATCH   = 'PATCH'.freeze
-  DELETE  = 'DELETE'.freeze
-  HEAD    = 'HEAD'.freeze
-  OPTIONS = 'OPTIONS'.freeze
-  LINK    = 'LINK'.freeze
-  UNLINK  = 'UNLINK'.freeze
-  TRACE   = 'TRACE'.freeze
-
-  # Rack environment variables
-  RACK_VERSION                        = 'rack.version'.freeze
-  RACK_TEMPFILES                      = 'rack.tempfiles'.freeze
-  RACK_ERRORS                         = 'rack.errors'.freeze
-  RACK_LOGGER                         = 'rack.logger'.freeze
-  RACK_INPUT                          = 'rack.input'.freeze
-  RACK_SESSION                        = 'rack.session'.freeze
-  RACK_SESSION_OPTIONS                = 'rack.session.options'.freeze
-  RACK_SHOWSTATUS_DETAIL              = 'rack.showstatus.detail'.freeze
-  RACK_MULTITHREAD                    = 'rack.multithread'.freeze
-  RACK_MULTIPROCESS                   = 'rack.multiprocess'.freeze
-  RACK_RUNONCE                        = 'rack.run_once'.freeze
-  RACK_URL_SCHEME                     = 'rack.url_scheme'.freeze
-  RACK_HIJACK                         = 'rack.hijack'.freeze
-  RACK_IS_HIJACK                      = 'rack.hijack?'.freeze
-  RACK_HIJACK_IO                      = 'rack.hijack_io'.freeze
-  RACK_RECURSIVE_INCLUDE              = 'rack.recursive.include'.freeze
-  RACK_MULTIPART_BUFFER_SIZE          = 'rack.multipart.buffer_size'.freeze
-  RACK_MULTIPART_TEMPFILE_FACTORY     = 'rack.multipart.tempfile_factory'.freeze
-  RACK_REQUEST_FORM_INPUT             = 'rack.request.form_input'.freeze
-  RACK_REQUEST_FORM_HASH              = 'rack.request.form_hash'.freeze
-  RACK_REQUEST_FORM_VARS              = 'rack.request.form_vars'.freeze
-  RACK_REQUEST_COOKIE_HASH            = 'rack.request.cookie_hash'.freeze
-  RACK_REQUEST_COOKIE_STRING          = 'rack.request.cookie_string'.freeze
-  RACK_REQUEST_QUERY_HASH             = 'rack.request.query_hash'.freeze
-  RACK_REQUEST_QUERY_STRING           = 'rack.request.query_string'.freeze
-  RACK_METHODOVERRIDE_ORIGINAL_METHOD = 'rack.methodoverride.original_method'.freeze
-  RACK_SESSION_UNPACKED_COOKIE_DATA   = 'rack.session.unpacked_cookie_data'.freeze
-
```

Comment:
Many of these constants are used by other libs, middleware, and apps. A quick GitHub search for for PATH_INFO, for example:

- https://github.com/tk-hamaguchi/rack_request_builder/blob/a7b634413162b02a238f7482a6760b8580aa3c25/lib/rack_request_builder/rack.rb#L1-L24
- https://github.com/rack-app/rack-app/blob/3347f38bd4767dd3be3484aac8a6a1378448a9b3/lib/rack/app/request_configurator.rb#L12-L16
- https://github.com/ioquatix/utopia/blob/9e016656353bd2884a11e7122ed6d305c62988b2/spec/utopia/session_spec.ru#L4-L20
- https://github.com/literalice/rack-pushstate-static/blob/c69ef8ae388785bdfcc1544b39465b94e99e902e/lib/rack/pushstate_static.rb#L26-L45